operator/internal/webhook/admission/pcs/authorization/handler.go

```diff
@@ -18,18 +18,144 @@ package authorization
+	}
+	resourceObjectKey := client.ObjectKeyFromObject(resourcePartialObjMeta)
+
+	// Get the parent PodCliqueSet
+	pcsPartialObjectMetadata, warnings, err := h.getParentPCSPartialObjectMetadata(ctx, resourcePartialObjMeta.ObjectMeta)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err).WithWarnings(warnings...)
+	}
+	// PCS can be nil if the resource has no reference of a PodCliqueSet or the referenced PodCliqueSet is not found.
+	if pcsPartialObjectMetadata == nil {
+		return admission.Allowed(fmt.Sprintf("admission allowed, no PodCliqueSet could be determined for resource: %v", resourceObjectKey)).WithWarnings(warnings...)
+	}
+
+	// Check if protection of PodCliqueSet managed resources has been disabled.
+	if metav1.HasAnnotation(pcsPartialObjectMetadata.ObjectMeta, apiconstants.AnnotationDisableManagedResourceProtection) {
+		log.Info("Resource has the \"grove.io/disable-managed-resource-protection\" annotation, authorized webhook bypassed. Admitting request.", "objectKey", resourceObjectKey)
+		return admission.Allowed(fmt.Sprintf("admission allowed, resource protection is disabled for PodCliqueSet: %v", client.ObjectKeyFromObject(pcsPartialObjectMetadata)))
+	}
+
+	switch req.Operation {
+	case admissionv1.Create, admissionv1.Update:
+		return h.handleCreateOrUpdate(req, resourceObjectKey)
+	case admissionv1.Delete:
+		return h.handleDelete(req, resourceObjectKey)
+	default:
+		return admission.Denied("Unhandled operation. Admission denied.")
+	}
+}
+
+// handleCreateOrUpdate allows creation/updation of managed resources only if the request is from the reconciler service account, or one of the exempted service accounts.
+func (h *Handler) handleCreateOrUpdate(req admission.Request, resourceObjectKey client.ObjectKey) admission.Response {
+	if req.UserInfo.Username == h.config.ReconcilerServiceAccountUserName {
+		return admission.Allowed(fmt.Sprintf("admission allowed, creation/updation of resource: %v is initiated by the grove reconciler service account", resourceObjectKey))
+	}
+
+	if slices.Contains(h.config.ExemptServiceAccountUserNames, req.UserInfo.Username) {
+		return admission.Allowed(fmt.Sprintf("admission allowed, creation/updation of resource: %v is initiated by exempt service account", resourceObjectKey))
+	}
+
+	return admission.Denied(fmt.Sprintf("admission denied, creation/updation of resource: %v is not allowed", resourceObjectKey))
+}
+
+// handleDelete allows deletion of managed resources only if the request is either from the service account used by the reconcilers
+// or the request is coming from one of the exempted service accounts.
+// There is an exception to this, where no restriction are placed on Pods.
+func (h *Handler) handleDelete(req admission.Request, resourceObjectKey client.ObjectKey) admission.Response {
+	if groupKindFromRequest(req) == podGVK.GroupKind() {
+		return admission.Allowed("admission allowed, deletion of resource: Pod is allowed for all users")
```

Comment:
message should indicate that deletion of pod is allowed by users having required RBAC. It is not really `all users`